core/engine.py

```
@@ -1,46 +1,111 @@
+        request = self.normalize_generation_request(request)
+        items: list[PackItemResult] = []
+
+        for index, sticker_input in enumerate(request.sticker_inputs):
+            try:
+                sticker = await self._media_normalizer.normalize_sticker(
+                    sticker_input,
+                    capabilities=capabilities,
+                )
+                items.append(PackItemResult(index=index, success=True, sticker=sticker))
+            except Exception as exc:
+                items.append(PackItemResult(index=index, success=False, error=str(exc)))
```

Ruff (0.15.7)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)